src/compas_timber/connections/solver.py

```diff
@@ -133,7 +135,7 @@ def find_topology(self, beam_a, beam_b, tol=TOLERANCE, max_distance=None):
 
         """
 
-        tol = self.TOLERANCE  # TODO: change to a unit-sensitive value
+        tol = ConnectionSolver.TOLERANCE  # TODO: change to a unit-sensitive value
```

Comment:
as we discussed, I think this solver (and especially future solvers) could indeed have state.
For example, this solver can get `tol` and `max_distance` upon instantiation. these don't really change throughout the run of the solver and thus can be instance state.

Reply:
good point. I will revert to an instance method

Reply:
should all the methods be instance methods then?